qiskit/circuit/__init__.py

```diff
@@ -54,81 +78,825 @@
+    <qiskit-transpiler>` on an *abstract circuit* that you constructed.
+
+physical qubit
+    A controllable two-level quantum system.  This is literally one "physics" qubit, such as a
+    transmon or the electronic state of a trapped ion.  A QPU may expose this directly as its
+    *hardware qubit*, or combine several physical qubits into a *logical qubit*.
+
+quantum processing unit (QPU)
+    Analogous to a CPU in classical computing or a GPU in graphics processing, a QPU is the hardware
+    that runs quantum operations on quantum data.  You can always expect a QPU that uses the
+    *circuit* model of computation to be able to perform some set of *gates*, and *measurement*
+    operations.  Depending on the particular technology, they also may be able to run some near-time
+    classical computations as well, such as classical control flow and bitwise calculations on
+    classical data.
+
+qubit
+    The basic unit of quantum information.
+
+unitary operation
+    A reversible operation on a quantum state. All quantum *gates* are unitary operations (by
+    definition).
+
+virtual qubit
+    An abstract, mathematical *qubit* used to build an *abstract circuit*.  Virtual qubits are how
+    one typically thinks about quantum algorithms at a high level; we assume that all quantum gates
+    are valid on all virtual qubits, and all virtual qubits are always connected to every other
+    virtual qubit.
+
+    When mapping to hardware, virtual qubits must be assigned to *hardware qubits*.  This mapping
+    need not be one-to-one.  Typically, one virtual qubit will need to be swapped from one hardware
+    qubit to another over the course of a circuit execution in order to satisfy coupling constraints
+    of the underlying QPU.  It is not strictly necessary for all virtual qubits used in a circuit to
+    be mapped to a physical qubit at any given point in a *physical circuit*; it could be that a
+    virtual qubit is measured (collapsing its state) and then never used again, so a new virtual
+    qubit could take its place.  Evaluating these conditions to map a virtual circuit to a physical
+    circuit is the job of :ref:`Qiskit's transpilation package <qiskit-transpiler>`.
+
+
+.. _circuit-module-api:
+
+API overview of :mod:`qiskit.circuit`
+=====================================
+
+All objects here are described in more detail, and in their greater context in the following
+sections.  This section provides an overview of the API elements documented here.
+
+..
+    TODO: actually write the "in-depth section on building circuits and cross-ref to it.
+
+The principal class is :class:`.QuantumCircuit`, which has its own documentation page, including
+an in-depth section on building circuits. The simplest quantum and classical data is built out of
+"bits" and "registers":
+
+* :class:`Bit`, an atom of data
+    * :class:`Qubit`
+    * :class:`Clbit`
+    * :class:`AncillaQubit`
```

Comment:
```suggestion
    * :class:`Qubit` - a quantum bit of quantum data
    * :class:`Clbit` - a classical bit (0 or 1, False or True) of classical data
    * :class:`AncillaQubit` - a quantum bit that provides 'scratch' space, e.g. for error correction or for more efficient realizations of gates
```

Reply:
I'm not clear this adds anything - we already defined the terms. We might want to add "ancilla" to the definitions, but the bits about qubit and clbit are repeating basic information we already gave.

Reply:
True, I was thinking about suggesting to remove AncillaQubits and AncillaRegister all together :-)

Reply:
Yeah, for sure we don't actually make any use of them right now.  We'd have to have a separate discussion for that, though, and they wouldn't be eligible for removal until Qiskit 2.0, and we'd still need to document them in the mean time.

Reply:
I added a brief definition of "ancilla qubit" to the definitions list in 2c1cc17232a, which hopefully is enough for this?